File: jdroid-java-core/src/main/java/com/jdroid/java/utils/IdGenerator.java
package com.jdroid.java.utils;

public class IdGenerator {
	
	private static Integer ID = 10000;
	
	public static synchronized Long getLongId() {
		ID++;
		return ID.longValue();
	}
	
	public static synchronized Integer getIntId() {
		return ID++;
	}
	

}

File: jdroid-java-core/src/main/java/com/dipien/java/utils/IdGenerator.kt
package com.dipien.java.utils

object IdGenerator {

    private var ID: Int = 10000

    @Synchronized
    fun getLongId(): Long {
        ID++
        return ID.toLong()
    }

    @Synchronized
    fun getIntId(): Int {
        return ID++
    }
}
